packages/domain/src/use-cases/inbox/DisableConversations.ts

```diff
@@ -0,0 +1,11 @@
+export interface IDisableConversationsGateway {
+  disableConversations(): Promise<void>;
+}
+
+export class DisableConversations {
+  constructor(private readonly gateway: IDisableConversationsGateway) {}
+
+  async execute(): Promise<void> {
+    await this.gateway.disableConversations();
```

Comment:
It seems the separate use case could be error prone leaving the consumer app in a dangerous state from user's privacy perspective.

I would simply include this call in the existing `WalletLogout` use case and remove the need for extra hook, documentation, example, etc.

Reply:
I agree, and I started implementing this, the problem is that conversationGateway requires `IConversationProvider` which we create on the fly in each `use<*>Conversation` hook, but we don't have it in `useWalletLogout` 😟